### ISSUE TITLE
Save File Formatting %date% evaluation in Batch Queue < 1 not using time of file creation

### ISSUE THREAD

**Issue by masslevel** [comfyanonymous/ComfyUI#1176]
If you use the Save File Formatting feature in the native ComfyUI Save Image node with a syntax like `%date:yyyyMMddhhmmss%` and in a Queue with a Batch Count < 1, all filenames will receive the same populated %date% variable.

## Example
In a queue with a batch count of < 1, all files will be named with the populated `%date%` variable of the first file indepenent of when the image file was created or the job in the queue ran.
```
myfile-20230809195712_00001_
myfile-20230809195712_00002_
myfile-20230809195712_00003_
myfile-20230809195712_00004_
...
```

I suppose `%date%` is evaluated when all jobs in a queue are created and maybe this is a limitation of how jobs are processed in general.

Would it be possible to have the Save File Formatting variables, especially `%date%`, be populated on either of the start of the current image generation in the queue or on file creation?

## Additional issues manifesting because of this
If you move or delete an image while a Queue job with a Batch count < 1 is running, the next filename that will be created fills that position. 

It continuous incrementing once all missing positions in the current *numbering* have been filled. Because of this the files will no longer be sorted chronologically as they were created.

## Example

List of images in the `/output/` directory
```
myfile-20230809195712_00001_
myfile-20230809195712_00002_
myfile-20230809195712_00003_
myfile-20230809195712_00004_
...
```
If you now delete or move the file `myfile-20230809195712_00003_`, the next generated image will fill its place with the same name.
Now you have 2 images with the file name.

Solving this would make organizing and sorting of image files more robust.

**Comment by xwev** [comment]
Yes, I second this. (I think you mean when batch queue > 1 (greater than 1), no?)

%date% should be read when job *starts* (or maybe when the image is actually saved), but not when enqueued. The current behavior makes it difficult to properly sort and compare output files (eg when %date% is used as a filename prefix). And I don't think the current behavior (%date% string assigned when job is enqueued) is what a user would expect.

**Comment by masslevel** [comment]
> (I think you mean when batch queue > 1 (greater than 1), no?)

Oh - my mistake. Thanks for the hint! I updated the title and text accordingly. I experimented with my own timedate function in `nodes.py` including the ability to add a suffix as well, but a native solution for this would still be really great to have.

**Comment by devingfx** [comment]
Hi !
Could you please point me out to the place where to find all possible naming variables like `%date%`? :pray:

**Comment by FeeZotter** [comment]
> Hi ! Could you please point me out to the place where to find all possible naming variables like `%date%`? 🙏

https://blenderneko.github.io/ComfyUI-docs/Interface/SaveFileFormatting/

**Comment by baoanhng** [comment]
This is an issue. I want it to record when the process starts, not when it inits.

**Comment by Webtransformer** [comment]
+1
